### ISSUE TITLE
Why does the average negative distance decreases during the training process?

### ISSUE THREAD

**Issue by SongZRui** [filipradenovic/cnnimageretrieval-pytorch#41]
I guess I am making mistakes somewhere, because I got the average distance decreases during the training process, and the test mAP does not keep increasing with the epoch like you showed in your paper though the training loss is decreasing. The learning rate and weight decay are set to default. The difference is that  I replaced the max pooling layers in VGG16 to a learnable mixed max and average pooling. Should I adjust the learning rate?
Many Thx in advance.

**Comment by filipradenovic** [comment]
The distance of positive pairs should be decreasing, while the distance of negative pairs should be increasing. You can try different learning rate values to see if something changes. I am not sure what could help in your particular case.

**Comment by QingSimon** [comment]
> I guess I am making mistakes somewhere, because I got the average distance decreases during the training process, and the test mAP does not keep increasing with the epoch like you showed in your paper though the training loss is decreasing. The learning rate and weight decay are set to default. The difference is that I replaced the max pooling layers in VGG16 to a learnable mixed max and average pooling. Should I adjust the learning rate?
> Many Thx in advance.
Hi SongZRui, have you solved the problem? I got the same results as you.
